extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoReactiveContextProvider.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.mongodb.runtime;
+
+import org.reactivestreams.Subscriber;
+
+import com.mongodb.RequestContext;
+import com.mongodb.reactivestreams.client.ReactiveContextProvider;
+
+import io.opentelemetry.context.Context;
+
+public class MongoReactiveContextProvider implements ReactiveContextProvider {
+
+    @Override
+    public RequestContext getContext(Subscriber<?> subscriber) {
+        return new MongoRequestContext(Context.current());
```

Comment:
When will these objects be removed?

Reply:
@brunobat removed from where? The mongo driver [docs ](https://mongodb.github.io/mongo-java-driver/5.1/apidocs/mongodb-driver-reactivestreams/com/mongodb/reactivestreams/client/ReactiveContextProvider.html) don't specify the behaviour. I'd expect it to be garbage collected after [the query has completed.](https://github.com/mongodb/mongo-java-driver/blob/84247d3be14f7f78be49e4596eae18c8e94c3f0a/driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/OperationExecutorImpl.java#L84)

We could however delete the current context right after retrieval from the `MongoRequestContext` in the `MongoTracingCommandListener`:
```
  @Override
    public void commandStarted(CommandStartedEvent event) {
        LOGGER.tracef("commandStarted event %s", event.getCommandName());

        Context parentContext = Optional.ofNullable(event.getRequestContext())
                .map(rc -> {
                    var ctx = (Context) rc.get(MongoRequestContext.OTEL_CONTEXT_KEY);
                    rc.delete(MongoRequestContext.OTEL_CONTEXT_KEY);
                    return ctx;
                })
                .orElseGet(Context::current);
```

Reply:
Please do, better safe than sorry.

Reply:
done